src/toady/bulk_reply_resolve_service.py

```
@@ -117,24 +158,56 @@
             elif atomic:
-                return self._perform_atomic_operations(operations)
+                assert transaction_id is not None  # Should be set for non-dry-run
+                return self._perform_atomic_operations_with_transaction(
+                    operations, transaction_id
+                )
             else:
-                return self._perform_non_atomic_operations(operations)
+                assert transaction_id is not None  # Should be set for non-dry-run
+                return self._perform_non_atomic_operations_with_transaction(
+                    operations, transaction_id
+                )
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "return", remove the leading "el" from "elif"

(R1705)